str_analysis/call_non_ref_pathogenic_motifs.py

```diff
@@ -0,0 +1,978 @@
+
+    final_height = (end1_y - start1_y) + (end2_y - start2_y) + 150
+
+    with open(output_file_path, "wt") as f:
+        f.write(f"""<svg width="{final_width}" height="{final_height}" xmlns="http://www.w3.org/2000/svg">""")
+        f.write(defs)
+        f.write(short_allele_contents)
+        # shift the long allele panel vertically so it appears just below the short allele panel
+        f.write("""<g transform="translate(0,%s)">""" % (end1_y - start2_y + 50))
+        f.write(long_allele_contents)
+        f.write("</g>")
+        f.write("</svg>")
+
+    return output_file_path
+
+
+def run_expansion_hunter_denovo(args):
+    """Run ExpansionHunterDenovo.
+
+    Arguments:
+        args (object): command-line arguments from argparse
+
+    Return:
+        Path of ExpansionHunterDenovo *.str_profile.json output file.
+    """
+
+    print("--"*10)
+    print(f"Running ExpansionHunterDenovo on {args.sample_id}")
+    output_prefix = f"{args.sample_id}.expansion_hunter_denovo"
+    expansion_hunter_denovo_command = f"""{args.expansion_hunter_denovo_path} profile \
+--reference "{args.reference_fasta}" \
+--reads "{args.bam_or_cram_path}" \
+--output-prefix "{output_prefix}"
+"""
+
+    run(expansion_hunter_denovo_command, verbose=args.verbose)
+
+    output_path = f"{output_prefix}.str_profile.json"
+    if not os.path.isfile(output_path):
+        print(f"ERROR: ExpansionHunterDenovo didn't produce a {output_path} file.")
+        return None
+
+    return output_path
+
+
+def process_reads_in_locus(
+        bam_or_cram_path,
+        reference_fasta,
+        locus_chrom,
+        locus_start_0based,
+        locus_end,
+        motif_size):
+    """Parses reads from the given bam/cram file that are within a window around the given STR locus. Then it computes
+    a list that, for each well-aligned read that overlaps the locus, contains the subsequence of bases from that read
+    that fall between the locus start and end coordinates. Also, it returns the total number of well-aligned read bases
+    to the left and to the right of the STR locus.
+
+    Args:
+        bam_or_cram_path (str): bam or cram path
+        reference_fasta (str): reference fasta path
+        locus_chrom (str): chromosome name of STR locus
+        locus_start_0based (int):  0-based start coordinate of STR locus
+        locus_end (int): end coordinate of STR locus
+        motif_size (int): motif size
+
+    Return:
+          overlapping_sequences, left_flank_n_well_aligned_bases, right_flank_n_well_aligned_bases
+    """
+    with pysam.Samfile(bam_or_cram_path, reference_filename=reference_fasta) as f:
+
+        # count reads in the left & right flanks to estimate read depth
+        # NOTE: f.fetch retrieves all reads that *overlap* the given interval
+        left_flank_n_well_aligned_bases = sum((r.query_alignment_length for r in f.fetch(
+            locus_chrom,
+            locus_start_0based - MARGIN - FLANK_SIZE,
```

Comment:
If this was a library with functions that could be used externally, I'd agree, but since these values are not expected to vary across different use-cases, I don't see the advantage of making them args.